easybuild/easyblocks/g/gamess_us.py

```diff
@@ -248,28 +250,43 @@ def test_step(self):
                 ])
 
             # run all exam<id> tests, dump output to exam<id>.log
-            n_tests = 47
-            for i in range(1, n_tests+1):
-                test_cmd = ' '.join(test_env_vars + [rungms, 'exam%02d' % i, self.version, '1', '2'])
-                (out, _) = run_cmd(test_cmd, log_all=True, simple=False)
-                write_file('exam%02d.log' % i, out)
+            total_num_tests = 47
+            badtests = []
+            if self.cfg['ddi_comm'] == 'mpi':
+                badtests = [5, 32, 39, 42, 45, 46, 47]
+            for i in range(1, total_num_tests+1):
+                runthistest = True
+                if self.cfg['ddi_comm'] == 'mpi':
+                    # Not all tests can run in parallel. Skip those that can't.
+                    if i in badtests:
+                        runthistest = False
```

Comment:
you can use `continue` instead of playing with a boolean variable like `runthistest`, which will just jump to the next iteration of the for loop (https://docs.python.org/2/reference/simple_stmts.html#the-continue-statement)

For the tests that can't be run in parallel, can we run them serially instead?


```python
skip_tests = []
if self.cfg['ddi_comm'] == 'mpi':
    # FIXME: reference to which tests are serial/parallel
    skip_tests = [5, 32, 39, 42, 45, 46, 47]
for i in range(range(1, total_num_tests+1):
    ncpus = 1
    if self.cfg['ddi_comm'] == 'mpi':
        ncpus = 2
        if i in skip_tests:
            # skip this tests & move to the next one
            
            continue
```